weather.py

```diff
@@ -0,0 +1,344 @@
+import argparse
+import calendar
+import csv
+import os
+
+required_labels = [
+    "PKT", "PKST", "Max TemperatureC", "Min TemperatureC", "Mean Humidity"
+]
+
+
+def read_weather_file(weather_data_file):
+    """
+    Opens a CSV weather data file and reads the data into a list of dictionaries,
+    filtering out unnecessary keys based on required_labels.
+
+    Args:
+    - weather_data_file (str): Path to the weather data file in CSV format.
+
+    Returns:
+    - list: A list of dictionaries, where each dictionary represents a day's
+      weather record for a month. Each dictionary contains only the keys specified
+      in required_labels.
+    """
+    month_weather_data = []
+    if not os.path.isfile(weather_data_file):
+        return month_weather_data
+
+    with open(weather_data_file, mode="r") as file:
+        csv_reader = csv.DictReader(file)
+        headers = [header.strip() for header in csv_reader.fieldnames]
+        for daily_weather_record in csv_reader:
+            daily_weather_record = {
+                key.strip(): value
+                for key, value in daily_weather_record.items()
+            }
+            filtered_daily_weather_record = {
+                key: daily_weather_record[key]
+                for key in required_labels
+                if key in headers
+            }
+            month_weather_data.append(filtered_daily_weather_record)
+
+    return month_weather_data
+
+
+def load_year_weather_data(weather_files_directory, year):
+    """
+    Loads weather data for a specified year from multiple monthly files.
+
+    Args:
+    - weather_files_directory (str): Directory path where weather files are stored.
+    - year (int): Year for which weather data is to be loaded.
+
+    Returns:
+    - list: Combined list of weather records for the entire year.
+    """
+    year_weather_data = []
+
+    for month_abbr in calendar.month_abbr[1:]:
+        month_weather_data = load_month_weather_data(weather_files_directory, year, month_abbr)
+        year_weather_data.extend(month_weather_data)
+
+    return year_weather_data
+
+
+def load_month_weather_data(weather_files_directory, year, month):

```

Comment:
Loading data should be irrespective of month or year. It should be dependent on size of input based on year or month. We need to make it abstract.